store/cifar10.py

```diff
@@ -1,39 +1,125 @@
-        if not Path(data_file_path).exists():
-            os.mkdir(data_file_path)
-        data_file_path += self.dataset_name + '.npy'
-        with Path(data_file_path).open('ab') as f:
-            for root, sub_dirs, files in os.walk(self.dataset_dir):
-                for file in files:
-                    if "data_batch" in file:
-                        path = self.dataset_dir +'/'+ file
-                        fo = open(path, 'rb')
-                        p = pickle.load(fo, encoding='bytes')
-                        data = np.array(p.get(b'data'))
-                        labels = np.array(p.get(b'labels'))
-                        data_points= []
-                        for data_point, label in zip(data,labels):
-                            data_points.append([data_point, label])
-                        nparr = np.array(data_points)
-                        np.save(f,nparr)
+        data_folder_path = self.get_data_folder_path()
+        if not Path(data_folder_path).exists():
+            print("Creating directory(s)", data_folder_path)
+            Path(data_folder_path).mkdir(parents=True, exist_ok=True)
+
+        # Create train and test directories
+        train_folder_path = data_folder_path + '/' + self.TRAIN_FOLDER
+        test_folder_path = data_folder_path + '/' + self.TEST_FOLDER
+        for path in [train_folder_path, test_folder_path]:
+            if not Path(path).exists():
+                print("Creating directory", path)
+                Path(path).mkdir(parents=True, exist_ok=True)
+
+        # Create the train data file
+        train_file_path = train_folder_path + '/' + self.DATA_FILE.format(0)
+        if Path(train_file_path).exists():
+            msg = "File " + train_file_path + "exists. "
+            if self.delete_existing:
+                msg += "Deleting it."
+                print(msg)
+                os.remove(train_file_path)
+            else:
+                msg += "Skipping."
+                print(msg)
+                return
+
+        # Generate train dataset
+        f_train = Path(train_file_path).open('ab')
+        for root, sub_dirs, files in os.walk(self.input_data_folder):
+            for file in files:
+                if "data_batch" not in file:
+                    continue
+                full_path = self.input_data_folder +'/'+ file
+                nparr = self.read_cifar_batch(full_path)
+                np.save(f_train, nparr)
+
+        # Generate test dataset
+        test_file_path = test_folder_path + '/' + self.DATA_FILE.format(0)
+        f_test = Path(test_file_path).open('ab')
+        test_data_file = self.input_data_folder + "/test_batch"
+        nparr = self.read_cifar_batch(test_data_file)
+        np.save(f_test, nparr)
+
+        # Create metadata for train and test folders
+        self.write_metadata()
+
+    def read_cifar_batch(self, batch_filename):
```

Comment:
Can you change this function name to parse_cifar_data? It is confusing whether we are reading from the dataset directory or getting the batch from shared memory?

Reply:
Will do.

Reply:
You are saving each batch file separately as .npy file? i.e for this cifar-10 there will be 5 .npy files?

Reply:
yeah